src/ring_theory/ideal/operations.lean

```diff
@@ -365,6 +365,45 @@ lemma mul_eq_bot {R : Type*} [integral_domain R] {I J : ideal R} :
     or.resolve_left (mul_eq_zero.mp ((I * J).eq_bot_iff.mp hij _ (mul_mem_mul hi hj))) ne0)),
  λ h, by cases h; rw [← ideal.mul_bot, h, ideal.mul_comm]⟩
 
+/-- A product of ideals in an integral domain is zero if and only if one of the terms
+is zero. Decidability is required for mutliset.erase-/
+lemma prod_eq_bot {R : Type*} [integral_domain R] [decidable_eq (ideal R)]
+  {s : multiset (ideal R)} : s.prod = ⊥ ↔ ∃ I ∈ s, I = ⊥ :=
```

Comment:
I feel like we should prove this for the product of two ideals, and then try to derive this from general stuff about `no_zero_divisors`.

Reply:
Well, indeed the case for two ideals was already there (it is `mul_eq_bot` just before the lemma I suggest to add) and the proof is an induction on that lemma to treat the general case.

Reply:
Right, but I think that induction has already been proved before... so my hope is that we can prove your lemma in 1 or 2 lines.

Reply:
More concretely, the first tactic can be `refine multiset.induction _ (λ J s ih, _) s` instead of induction on the cardinality.

Reply:
@Vierkantor Thanks, I thought that this was what @jcommelin was suggesting and I am now trying to implement this. I changed the label to WIP.

Reply:
Actually, my hope is that `ideal R` will be some sort of semiring with no zero divisors, and so you should be able to prove this in 1 line using a lemma from the library.
It's still a useful lemma, because it is stated using `\bot` instead of `0`.

Reply:
I might be wrong, but I'd say that my lemma is somehow the proof that `ideal R` is a semiring with no zero divisors when `R` is an integral domain. It seems strange to me that we can have that result without this lemma (which is not to say that my proof is nice, or that it cannot be cut by 99%, as I am trying to do...)

Reply:
No, you only need the lemma for the product of two ideals. After that, the general machine should be able to take over, so that you don't have to do a 15-line induction proof all over again.

Reply:
I see, but after some trials I was unable to put your suggestion into effect. I golfed my old proof so that it is 10 lines shorter, but it is still not as you wanted it. I will label it back to `awaiting_review` hoping to get some hints if in the present form this is still unacceptable.

Reply:
I pushed a sketch of what I mean

Reply:
It seems that the lemma that I was thinking of doesn't exist yet, so I pushed its statement as `move_this`. It should go to another file. And there should probably also be versions for `list.prod` and `finset.prod`.

Reply:
I see now what you meant: thanks! Concerning the `move_this`, do you think that it should "roughly" be proved along the lines of my old proof (replacing ideals by members of `R`)? In that case, I can try to move it adding my proof, otherwise I'd rather kill this `PR`.

Reply:
Yes, that's exactly what I had in mind. In practice, what people usually do is first prove the version for `list.prod`, and then use that for the `multiset.prod` version. That is probably the shortest path.

Reply:
Alles klar! I am slightly busy finishing another project now, so I will put an awaiting-author label and will work on this ASAP. Thanks.